typescript/sdk/src/token/EvmWarpCrudModule.ts

```diff
@@ -0,0 +1,216 @@
+    return this.reader.deriveWarpRouteConfig(
+      this.args.addresses.deployedWarpRoute,
+    );
+  }
+
+  /**
+   * Updates the Warp Route contract with the provided configuration.
+   *
+   * @remark Currently only supports updating ISM or hook
+   *
+   * @param config - The configuration for the token router to be updated.
+   * @returns An array of Ethereum transactions that were executed to update the contract, or an error if the update failed.
+   */
+  public async update(
+    config: DerivedTokenRouterConfig,
+  ): Promise<EthersV5Transaction[]> {
+    return [
+      ...(await this.updateIsm(config)),
+      ...(await this.updateHook(config)),
+    ];
+  }
+
+  /**
+   * Updates an existing Warp route ISM with a given configuration.
+   *
+   * This method handles two cases:
+   * 1. If the `config.interchainSecurityModule` is an object,
+   *  - Checks if the current onchain ISM configuration matches the provided configuration.
+   *  - If not, it deploys a new ISM module, and updates the contract's ISM.
+   * 2. If the `config.interchainSecurityModule` is a string
+   *  - It attempts to derive the ISM from the provided string, and updates the contract's ISM.
+   *
+   * @param config - The token router configuration, including the ISM configuration.
+   * @returns An array of Ethereum transactions that need to be executed to update the ISM configuration.
+   */
+  async updateIsm(
+    config: DerivedTokenRouterConfig,
+  ): Promise<EthersV5Transaction[]> {
+    const transactions: EthersV5Transaction[] = [];
+
+    const contractToUpdate = await this.args.addresses[config.type].deployed();
+
+    if (typeof config.interchainSecurityModule === 'string') {
+      // Derive & set ISM
+      const ism = await this.reader.evmIsmReader.deriveIsmConfig(
+        config.interchainSecurityModule,
+      );
+      transactions.push({
+        transaction:
+          await contractToUpdate.populateTransaction.setInterchainSecurityModule(
+            ism.address,
+          ),
+        type: ProviderType.EthersV5,
+      });
+    } else if (typeof config.interchainSecurityModule === 'object') {
+      const onchainConfig = await this.read();
+      if (
+        config.interchainSecurityModule.type !==
+        onchainConfig.interchainSecurityModule!.type
+      ) {
+        // Deploy & set ISM
+        const ismModule = await this.deployIsm(config);
+        transactions.push({
+          transaction:
+            await contractToUpdate.populateTransaction.setInterchainSecurityModule(
+              ismModule,
+            ),
+          type: ProviderType.EthersV5,
+        });
+      }
+    }
+
+    return transactions;
+  }
+
+  /**
+   * Deploys the ISM using the provided configuration.
+   *
+   * @param config - The configuration for the ISM to be deployed.
+   * @returns The deployed ISM contract address.
+   */
+  public async deployIsm(config: DerivedTokenRouterConfig): Promise<string> {
```

Comment:
This module is not meant to use to deploy an ISM, but it can deploy one by calling another module. I think inter-module calls are fine. open to suggestions